telescope.yaml

```diff
@@ -0,0 +1,121 @@
+# This file creates replicas number of Pods, all served with one Kubernetes Service resource
+# each pod has 2 containers (telescope and redis) as has been specified in following Deployment resource
+
+# Change <DockerID> with the Docker Hub namespace (ID/username) in this file
+# To deploy project on Kubernetes we must have telescope image from `Dockerfile-deploy` file pushed on Docker Hub (instruction in `Dockerfile-deploy`).
+# We need `kubectl` installed, and for local simulation of the deployment, `minikube` installed and running.
+# deploy the app using following command
+# $ kubectl create -f telescope.yaml
+# to watch the pods as they gets created
+# $ kubectl get pods --watch
+# to see stdout of containers (telescope or redis) in one specific instance (pod)
+# $ kubctl logs <pod name as shown by above command> -c [telescope | redis]   ---> ex. $ kubectl logs telescope-7b567445fc-pn7cb -c telescope
+# finally access to the app which is running in Kubernetes cluster using this command
+# $ minikube service telescope
+# we can scale the app using the following command (example: from 3 replicas to 5)
+# $ kubectl scale --replicas=5 telescope
+# re-creating and replacing resources defined in a YAML file (ONLY IN MINIKUBE):
+# $ kubectl replace --force -f telescope.yaml   -----> Will cause service outage; Do not use in real deployment in Kubernetes
+
+# Service resource
+## telescope needs a service resource to be accessible to the outside the cluster
+## Service resource also guarantees the same IP address after each restart
+
+# version of Kubernetes API- using stable version
+apiVersion: v1
+# resource type and label
+kind: Service
+metadata:
+  name: telescope
+# specifications of resource
+spec:
+  selector:
+    name: telescope
```

Comment:
I had to use```app: telescope``` here otherwise an endpoint wasn't being assigned to the service

Reply:
I didn't need to have a named endpoint. I tried to prevent any Kubernetes vocabulary that I thought it was unnecessary. Why do you think we need to have a named endpoint?

Reply:
I was never able to get the deployment and the service up and running through the yaml, whenever I tried using the service with ```minikube service telescope``` and I wouldn't be able to access the website

Until the change, these were my steps to get it running
1. ```kubectl create deployment -f telescope.yaml```
2. ```kubectl delete services telescope```
3. ```kubectl expose deployment telescope --type=NodePort```
or ```kubectl expose deployment telescope --type=LoadBalancer```
4. ```minikube service telescope```

Reply:
I was not able to get the localhost although the backend was working. So if you could get the localhost with this change, I should change it as well.